src/modules/products/controllers/search.controller.ts

```
@@ -0,0 +1,42 @@
+    @Query('fields') fields?: string,
+    @Query('category') category?: string,
+    @Query('minPrice') minPrice?: number,
+    @Query('maxPrice') maxPrice?: number,
+    @Query('brand') brand?: string,
+  ) {
+    try {
+      // Ensure `order` is either "ASC" or "DESC" (default to "ASC")
+      const validatedOrder: 'ASC' | 'DESC' = order === 'DESC' ? 'DESC' : 'ASC';
+
+      // Create DTO instances manually
+      const searchDto: SearchDto = { query, sortBy, order: validatedOrder, page, limit, fields };
```

ESLint

[error] Replace `·query,·sortBy,·order:·validatedOrder,·page,·limit,·fields` with `⏎········query,⏎········sortBy,⏎········order:·validatedOrder,⏎········page,⏎········limit,⏎········fields,⏎·····`

(prettier/prettier)